dev/breeze/src/airflow_breeze/commands/release_management_commands.py

```diff
@@ -2394,3 +2395,270 @@ def prepare_python_client(
+    "Chart.yaml, unless the --ignore-version-check flag is used.",
+    envvar="VERSION",
+)
+@click.option(
+    "--version-suffix",
+    help="Version suffix used to publish the package. Needs to be present as we always build "
+    "archive using release candidate tag.",
+    required=True,
+    envvar="VERSION_SUFFIX",
+)
+@click.option(
+    "--ignore-version-check",
+    is_flag=True,
+    help="Ignores result of version update check. Produce tarball regardless of "
+    "whether version is correctly set in the Chart.yaml.",
+)
+@click.option(
+    "--skip-tagging",
+    is_flag=True,
+    help="Skip tagging the chart. Useful if the tag is already created, or when you verify the chart.",
+)
+@click.option(
+    "--skip-tag-signing",
+    is_flag=True,
+    help="Skip signing the tag. Useful for CI where we just tag without signing the tag.",
+)
+@click.option(
+    "--override-tag",
+    is_flag=True,
+    help="Override tag if it already exists. Useful when you want to re-create the tag, usually when you"
+    "test the breeze command locally.",
+)
+@option_dry_run
+@option_verbose
+def prepare_helm_chart_tarball(
+    version: str | None,
+    version_suffix: str,
+    ignore_version_check: bool,
+    override_tag: bool,
+    skip_tagging: bool,
+    skip_tag_signing: bool,
+) -> None:
+    import yaml
+
+    chart_yaml_file_content = CHART_YAML_FILE.read_text()
+    chart_yaml_dict = yaml.safe_load(chart_yaml_file_content)
+    version_in_chart = chart_yaml_dict["version"]
+    airflow_version_in_chart = chart_yaml_dict["appVersion"]
+    airflow_version_from_pypi = _get_latest_airflow_version_from_pypi()
+    if ignore_version_check:
+        if not version:
+            version = version_in_chart
+    else:
+        if not version or not version_suffix:
+            get_console().print(
+                "[error]You need to provide --version and --version-suffix parameter unless you "
+                "use --ignore-version-check[/]"
+            )
+            sys.exit(1)
+    get_console().print(f"[info]Latest Airflow version from PyPI: {airflow_version_from_pypi}[/]")
+    get_console().print(f"[info]Airflow version in chart: {airflow_version_in_chart}[/]")
+    updating = False
+    if version_in_chart != version:
+        get_console().print(
+            f"[warning]Version in chart.yaml ({version_in_chart}) does not match the version "
+            f"passed as parameter ({version}). Updating[/]"
+        )
+        updating = True
+        chart_yaml_file_content = chart_yaml_file_content.replace(
+            f"version: {version_in_chart}", f"version: {version}"
+        )
+    else:
+        get_console().print(f"[success]Version in chart.yaml is good: {version}[/]")
+    if airflow_version_from_pypi != airflow_version_in_chart:
+        get_console().print(
+            f"[warning]Airflow version in chart.yaml ({airflow_version_in_chart}) does not match the "
```

Comment:
Right. Moment.

Reply:
More needs to happen than even just bumping the tag in values. If it were me, I'd just check that appVersion matches the tag/airflowVersion in values and if it fails, just bail out instead of trying to update all the things. My 2c.

Reply:
Added checking if "airflowVersion" and "defaultAirflowTag" are the same (if not - fail hard):

<img width="764" alt="Screenshot 2024-01-22 at 21 26 09" src="https://github.com/apache/airflow/assets/595491/82819edc-921d-4c10-816e-2230c7dd0ffe">


If they both match and Chart.yaml is different - update Chart.yaml with the right version and ask the release manager to make PR:

<img width="871" alt="Screenshot 2024-01-22 at 21 27 40" src="https://github.com/apache/airflow/assets/595491/b48d92e8-5d11-41d6-9703-1f8a257ca6f4">

Reply:
> I'd just check that appVersion matches the tag/airflowVersion in values and if it fails, just bail out instead of trying to update all the things 

I think the only case that we want to update things is when Chart.yaml has wrong appVersion - then I think this is not really a problem to update Chart.yaml ?

Reply:
Yeah if we just make sure appVersion matches the tag, good enough imo.

Reply:
Resolving then. Rebase to apply just merged main fix.

Reply:
Unresolving this. I still see it checking pypi, not values. Bad rebase?

Reply:
Indeed. Should be fine now.